pybamm/models/submodels/base_submodel.py

```diff
@@ -57,7 +57,11 @@ def __init__(
         self.name = name
 
         self.external = external
-        self.options = options or {}
+        self.options = options
+
+        # Save whether the submodel is a half-cell submodel
+        we = self.options.get("working electrode", "both")
```

Comment:
```suggestion
        try:
            we = self.options.get("working electrode", "both")
        except AttributeError:
            we = "both"
            
```
I think this should fix the error (probably the syntax can be improved though).